src/hotspot/share/memory/arena.cpp

```diff
@@ -91,9 +91,26 @@ class ChunkPool {
     _first = nullptr;
   }
 
-  static void clean() {
+  bool empty() const {
+    return _first == nullptr;
+  }
+
+  static bool needs_cleaning() {
     for (int i = 0; i < _num_pools; i++) {
-      _pools[i].prune();
+      if (!_pools[i].empty()) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
Something seems wrong here? Having only empty pools means that `::prune()` is a no-op.

Reply:
Thanks for catching this.

Reply:
Somewhat scary that we don't have a test for that.